src/nemos/basis/_identity.py

```diff
@@ -0,0 +1,183 @@
+from typing import Optional, Tuple
+
+import numpy as np
+from numpy._typing import ArrayLike
+from numpy.typing import NDArray
+from pynapple import Tsd, TsdFrame, TsdTensor
+
+from ..type_casting import support_pynapple
+from ..typing import FeatureMatrix
+from ._basis import Basis, check_fraction_valid_samples, check_transform_input
+from ._basis_mixin import AtomicBasisMixin
+
+
+class IdentityBasis(Basis, AtomicBasisMixin):
+    """
+    Base class for the Identity basis.
+
+    This basis should be used when one wants to incorporate raw inputs
+    as part of a composite basis.
+
+    Parameters
+    ----------
+    n_basis_funcs :
+        The number of basis functions.
+    label :
+        The label of the basis, intended to be descriptive of the task variable being processed.
+        For example: velocity, position, spike_counts.
+    """
+
+    def __init__(
+        self,
+        n_basis_funcs: int,
+        label: Optional[str] = None,
+    ) -> None:
+        AtomicBasisMixin.__init__(self, n_basis_funcs=n_basis_funcs)
+        super().__init__(
+            label=label,
+            mode="eval",
+        )
+        self._n_input_dimensionality = 1
+
+    @support_pynapple(conv_type="numpy")
+    @check_transform_input
+    def _evaluate(
+        self, sample_pts: ArrayLike | Tsd | TsdFrame | TsdTensor
+    ) -> FeatureMatrix:
+        """
+        Returns the samples as a 2D array.
+
+        Parameters
+        ----------
+        sample_pts:
+            Array of samples. The shape can be arbitrary, as long as the first axis is the
+            sample axis.
+
+        Returns
+        -------
+        :
+            The samples with an extra axis, the n_basis_funcs axis which is = 1.
+
+        """
+        vmin = np.nanmin(sample_pts, axis=0) if self.bounds is None else self.bounds[0]
+        vmax = np.nanmax(sample_pts, axis=0) if self.bounds is None else self.bounds[1]
+        sample_pts[(sample_pts < vmin) | (sample_pts > vmax)] = np.nan
+        check_fraction_valid_samples(
+            sample_pts,
+            err_msg="All the samples lie outside the [vmin, vmax] range.",
+            warn_msg="More than 90% of the samples lie outside the [vmin, vmax] range.",
+        )
+        return sample_pts[..., np.newaxis]
+
+    def evaluate_on_grid(self, n_samples: int) -> Tuple[Tuple[NDArray], NDArray]:
```

Comment:
I think return type is wrong? shouldn't it just be `Tuple[NDarray, NDArray]`?

Reply:
Yes!